src/languages/yaml.js

```diff
@@ -47,6 +47,22 @@ export default function(hljs) {
     ]
   };
 
+  // Strings inside objects can't start with { or } or [ or ] or ,
+  // and can't end with } or ] or ,
+  var INSIDE_OBJECT_STRING = {
+    className: 'string',
+    relevance: 0,
+    variants: [
+      {begin: /'/, end: /'/},
+      {begin: /"/, end: /"/},
+      {begin: /[^\s,{}[\]](\S*[^\s,\]}])?/}
```

Comment:
Why would a simple repeated negation not work:

```
[^\s,{}[\]]+
```

That's the simplest thing I came up with playing.

Reply:
I was attempting to still allow these characters in strings, just not as the first or last character.

Reply:
In my testing in Ruby at least that is not allowed.  The character means the end of the string.

Reply:
Ah, you're right, its the same way in Python. I guess I shouldn't have assumed. I'll use the repeated negation.

Reply:
Never assume with YAML. :-)  Although it's a good rule in other context also. :-)